ruby-call-signature/src/test/java/org/jetbrains/ruby/codeInsight/types/signature/MergeTest.kt

```diff
@@ -19,38 +26,33 @@ class MergeTest : TestCase() {
         assertEquals(serialized, testDataClean)
     }
 
-    private fun generateRTuple(args: List<String>, returnType: String): RTuple {
-        val gemInfo = GemInfo.Impl("test_gem", "1.2.3")
-        val classInfo = ClassInfo.Impl(gemInfo, "TEST1::Fqn")
-        val location = Location("test1test1", 11)
-        val methodInfo = MethodInfo.Impl(classInfo, "met1", RVisibility.PUBLIC, location)
-
-        val params = args.indices.map { ParameterInfo("a" + it, ParameterInfo.Type.REQ) }
-
-        return RTuple(methodInfo, params, args, returnType)
+    private fun checkSignaturesSerialization(signatures: List<SignatureInfo.Impl>,
+                                             newSignatures: List<SignatureInfo>, contractsTestData: List<String>) {
+        assertTrue(signatures.size == newSignatures.size)
+        for (i in 0 until newSignatures.size) {
+            assertTrue(signatures[i].methodInfo == newSignatures[i].methodInfo)
+            checkSerialization(signatures[i].contract, contractsTestData[i % 4])
+        }
     }
 
-    @Test
-    fun testSimpleMerge() {
-        val args1 = listOf("String1", "String2", "String3")
-        val args2 = listOf("Int1", "String2", "String3")
-        val args3 = listOf("String1", "Int2", "String3")
-        val args4 = listOf("Int1", "Int2", "String3")
 
-        val args5 = listOf("Int1", "Int2", "Int3")
+    private fun generateComplicatedContract() : RSignatureContract {
+        val args1 = listOf("a1", "c2", "a3", "a4")
+        val args2 = listOf("a1", "b2", "a3", "a4")
+        val args3 = listOf("a1", "c2", "b3", "a4")
+        val args4 = listOf("a1", "b2", "b3", "a4")
 
-        val testArgs1 = listOf("Int1", "Int2", "Int3")
-        val testArgs2 = listOf("String1", "Int2", "Int3")
```

Comment:
Serialisation has nothing to do with Merge, please move such tests to the corresponding class

Reply:
Ok. I also want to extract common stuff for *Info generation into BaseTestClass.

Reply:
Good idea. BTW there are already serialisation tests

Reply:
Please take note on `SignatureInfoData` and stuff: https://upsource.jetbrains.com/ruby-type-inference/file/1d921936421673c0375ac555967545959bfafbdf/storage-server-api/src/test/java/org/jetbrains/ruby/codeInsight/types/storage/server/impl/SignatureContractSerializationTest.kt?nav=964:981:focused&line=0

I mean they are not ideal, too and worth extracring to some testdata place.